### ISSUE TITLE
"keep_md: true" doesn't process 'r' code in header

### ISSUE THREAD

**Issue by nometus** [rstudio/rmarkdown#450]
originally reported to knitr yihui/knitr#1056

An example is the file below.  While the "knit HTML" button viewer displays the correct html output, the "test.md" file has the header "R" code `"`r Sys.Date()`"` unprocessed.  The rest of the "test.md" file (outside the header) look to process the r-code fine.

test.Rmd 

``````

---
title: "BB Test"
author: "BB"
date: "`r Sys.Date()`"
output: 
  html_document:
    keep_md: true

---

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r}
summary(cars)
```
``````

**Comment by jjallaire** [comment]
That's correct. We just pass the front matter straight through. There
unfortunately isn't an easy way to overcome this behavior given the
markdown toolchain that we are calling.

**Comment by yihui** [comment]
I proposed a fix at #1169.
